base/src/commonMain/kotlin/utils/Decimal.kt

```diff
@@ -19,27 +19,67 @@
 
 package com.splendo.kaluga.base.utils
 
-class DecimalException(message: String?) : ArithmeticException(message)
+import kotlin.math.pow
 
 /**
  * Immutable, arbitrary-precision signed decimal numbers.
  */
-expect class Decimal : Comparable<Decimal>
+sealed class Decimal : Comparable<Decimal> {
```

Comment:
would be nice to have equals + hashcode + tostring implemented

Reply:
Since this sealed class contains only objects/data classes, hashCode, equals and toString are already implemented